tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -227,6 +229,273 @@ class ZoneUnitarySysTest : public EnergyPlusFixture
+
+    // reset coil sizing for next test
+    WaterCoils::WaterCoil(CoilNum).DesAirVolFlowRate = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).DesInletAirTemp = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).DesOutletAirTemp = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).DesInletWaterTemp = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).DesInletAirHumRat = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).DesOutletAirHumRat = DataSizing::AutoSize;
+    WaterCoils::WaterCoil(CoilNum).MaxWaterVolFlowRate = DataSizing::AutoSize;
+
+    // size same coils in UnitarySystem
+    int AirLoopNum(1);
+    bool FirstHVACIteration(true);
+    UnitarySys thisSys;
+    UnitarySys *mySys(&thisSys);
+    mySys->m_CoolCoilExists = true;
+    mySys->m_HeatCoilExists = true;
+    mySys->m_MaxCoolAirVolFlow = DataSizing::AutoSize;
+    mySys->m_MaxHeatAirVolFlow = DataSizing::AutoSize;
+    mySys->m_CoolingSAFMethod = DataSizing::SupplyAirFlowRate;
+    mySys->m_HeatingSAFMethod = DataSizing::SupplyAirFlowRate;
+    mySys->m_DesignCoolingCapacity = DataSizing::AutoSize;
+    mySys->m_DesignHeatingCapacity = DataSizing::AutoSize;
+    mySys->m_CoolingCoilType_Num = DataHVACGlobals::Coil_CoolingWater;
+    mySys->m_CoolingCoilName = "Test Water Cooling Coil";
+    mySys->m_HeatingCoilType_Num = DataHVACGlobals::Coil_HeatingWater;
+    mySys->m_HeatingCoilName = "Test Water Heating Coil";
+    WaterCoils::GetWaterCoilsInputFlag = false;             // don't overwrite these coil data
+    WaterCoils::MySizeFlag = true;                          // need to size again for UnitarySystem
+    WaterCoils::WaterCoil(1).DesWaterCoolingCoilRate = 0.0; // reset these to be sure they get recalculated
+    WaterCoils::WaterCoil(2).DesWaterHeatingCoilRate = 0.0;
+    OutputReportPredefined::SetPredefinedTables();
+    DataGlobals::DoingSizing = true;
+
+    mySys->sizeSystem(FirstHVACIteration, AirLoopNum);
+
+    // Show coil sizes in UnitarySystem
+    EXPECT_NEAR(6672.0, WaterCoils::WaterCoil(1).DesWaterCoolingCoilRate, 1.0);
+    EXPECT_NEAR(1154.0, WaterCoils::WaterCoil(2).DesWaterHeatingCoilRate, 1.0);
+
+    // the water cooling coil sizes are only different by the air density used in capacity calculation
+    // water coils use StdRhoAir and UnitarySystem coils use actual air density
+    Real64 CoilInTemp = DataSizing::FinalSysSizing(1).MixTempAtCoolPeak;
+    Real64 CoilInHumRat = DataSizing::FinalSysSizing(1).MixHumRatAtCoolPeak;
+    Real64 rhoair = Psychrometrics::PsyRhoAirFnPbTdbW(DataEnvironment::StdBaroPress, CoilInTemp, CoilInHumRat);
+    // this is the ratio of UnitarySystem cooling coil size to coil-on-branch water cooling coil size
+    Real64 rhoRatio = DataEnvironment::StdRhoAir / rhoair;
+
+    EXPECT_NEAR(coil1CoolingCoilRate, rhoRatio * WaterCoils::WaterCoil(1).DesWaterCoolingCoilRate, 1.0);
+    EXPECT_NEAR(coil1CoolingCoilRate, rhoRatio * mySys->m_DesignCoolingCapacity, 1.0);
```

Comment:
water cooling coil on branch is same size as in UnitarySystem except for rho difference